client/src/main/java/io/pravega/client/batch/SegmentInputSplit.java

```diff
@@ -0,0 +1,59 @@
+/**
+ * Copyright (c) 2017 Dell Inc., or its subsidiaries. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ */
+package io.pravega.client.batch;
+
+import com.google.common.annotations.Beta;
+import com.google.common.base.Preconditions;
+import io.pravega.client.segment.impl.Segment;
+import lombok.AccessLevel;
+import lombok.Builder;
+import lombok.Data;
+import lombok.NonNull;
+import lombok.Setter;
+
+/**
+ * Class to represent a Segment split information.
+ */
+@Beta
+@Data
+@Builder
+public class SegmentInputSplit {
```

Comment:
We cannot use a Segment because we need to differentiate between a writeOffset and endOffset.
I had named this class as SegmentMetadata earlier. Need inputs to improve the name of the class.